--- firefox-48.0.1/old-configure.in.omv~	2016-08-18 23:18:25.642529946 +0200
+++ firefox-48.0.1/old-configure.in	2016-08-18 23:19:08.766274007 +0200
@@ -3108,7 +3108,7 @@ then
               AC_MSG_ERROR([$MOZ_QT_PKG_ERRORS Need qtbase development packages, (On Ubuntu, you might try installing the packages qtbase5-dev libqt5opengl5-dev.)])
             ])
             QT5INCDIR=`pkg-config --variable=includedir Qt5Gui`
-            MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QT5INCDIR/QtGui/$QT_VERSION/QtGui"
+            MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QT5INCDIR/QtGui/$QT_VERSION/QtGui -I$QT5INCDIR -I$QT5INCDIR/QtCore -I$QT5INCDIR/QtGui -I$QT5INCDIR/QtXml -I$QT5INCDIR/QtNetwork -I$QT5INCDIR/QtPrintSupport -I$QT5INCDIR/QtNetwork/$QT_VERSION/QtNetwork/private"
             if test "$NS_PRINTING"; then
                 PKG_CHECK_MODULES(MOZ_QT_WIDGETS, Qt5PrintSupport, ,
                 [
@@ -3137,7 +3137,8 @@ then
         5.*)
             AC_MSG_RESULT("Using qt5: $QT_VERSION")
             MOZ_QT_LIBS="$MOZ_QT_LIBS -L$QTDIR/lib/ -lQt5Gui -lQt5Network -lQt5Core -lQt5Xml"
-            MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QTDIR/include/QtGui/$QT_VERSION/QtGui"
+            QT5INCDIR=`pkg-config --variable=includedir Qt5Gui`
+            MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QT5INCDIR/QtGui/$QT_VERSION/QtGui -I$QT5INCDIR -I$QT5INCDIR/QtCore -I$QT5INCDIR/QtGui -I$QT5INCDIR/QtXml -I$QT5INCDIR/QtNetwork -I$QT5INCDIR/QtPrintSupport -I$QT5INCDIR/QtNetwork/$QT_VERSION/QtNetwork/private"
             if test "$NS_PRINTING"; then
                 MOZ_QT_LIBS="$MOZ_QT_LIBS -lQt5Widgets -lQt5PrintSupport"
                 MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QTDIR/include/QtPrintSupport"
--- firefox-48.0.1/old-configure.omv~	2016-08-18 23:16:38.058168813 +0200
+++ firefox-48.0.1/old-configure	2016-08-18 23:19:23.382187278 +0200
@@ -16257,7 +16257,7 @@ echo "configure:16229: checking MOZ_QT_L
   fi
 
             QT5INCDIR=`pkg-config --variable=includedir Qt5Gui`
-            MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QT5INCDIR/QtGui/$QT_VERSION/QtGui"
+            MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QT5INCDIR/QtGui/$QT_VERSION/QtGui -I$QT5INCDIR -I$QT5INCDIR/QtCore -I$QT5INCDIR/QtGui -I$QT5INCDIR/QtXml -I$QT5INCDIR/QtNetwork -I$QT5INCDIR/QtPrintSupport -I$QT5INCDIR/QtNetwork/$QT_VERSION/QtNetwork/private"
             if test "$NS_PRINTING"; then
                 succeeded=no
 
@@ -16443,7 +16443,8 @@ test -n "$HOST_RCC" || HOST_RCC=""""
         5.*)
             echo "$ac_t"""Using qt5: $QT_VERSION"" 1>&6
             MOZ_QT_LIBS="$MOZ_QT_LIBS -L$QTDIR/lib/ -lQt5Gui -lQt5Network -lQt5Core -lQt5Xml"
-            MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QTDIR/include/QtGui/$QT_VERSION/QtGui"
+            QT5INCDIR=`pkg-config --variable=includedir Qt5Gui`
+            MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QT5INCDIR/QtGui/$QT_VERSION/QtGui -I$QT5INCDIR -I$QT5INCDIR/QtCore -I$QT5INCDIR/QtGui -I$QT5INCDIR/QtXml -I$QT5INCDIR/QtNetwork -I$QT5INCDIR/QtPrintSupport -I$QT5INCDIR/QtNetwork/$QT_VERSION/QtNetwork/private"
             if test "$NS_PRINTING"; then
                 MOZ_QT_LIBS="$MOZ_QT_LIBS -lQt5Widgets -lQt5PrintSupport"
                 MOZ_QT_CFLAGS="$MOZ_QT_CFLAGS -I$QTDIR/include/QtPrintSupport"
--- firefox-48.0.1/widget/nsShmImage.h.omv~	2016-08-18 23:27:07.889434164 +0200
+++ firefox-48.0.1/widget/nsShmImage.h	2016-08-18 23:27:18.316372416 +0200
@@ -18,6 +18,7 @@
 #include "Units.h"
 
 #include <X11/Xlib.h>
+#include <X11/Xutil.h>
 #include <X11/extensions/XShm.h>
 
 class nsShmImage {
--- firefox-48.0.1/toolkit/components/remote/nsQtRemoteService.cpp.omv~	2016-08-18 23:51:09.400065396 +0200
+++ firefox-48.0.1/toolkit/components/remote/nsQtRemoteService.cpp	2016-08-18 23:51:18.522009043 +0200
@@ -83,7 +83,7 @@ nsQtRemoteService::Startup(const char* a
 }
 
 NS_IMETHODIMP
-nsQtRemoteService::RegisterWindow(nsIDOMWindow* aWindow)
+nsQtRemoteService::RegisterWindow(mozIDOMWindow* aWindow)
 {
   return NS_ERROR_NOT_IMPLEMENTED;
 }
--- firefox-48.0.1/gfx/thebes/gfxQPainterSurface.cpp.omv~	2016-08-18 23:59:01.648163858 +0200
+++ firefox-48.0.1/gfx/thebes/gfxQPainterSurface.cpp	2016-08-18 23:59:42.219915675 +0200
@@ -9,6 +9,7 @@
 #ifdef CAIRO_HAS_QT_SURFACE
 #include "cairo-qt.h"
 #include "gfxQPainterSurface.h"
+#include "../2d/HelpersCairo.h"
 
 gfxQPainterSurface::gfxQPainterSurface(QPainter *painter)
 {
