--- browser/app/profile/firefox.js	2012-06-01 12:03:47.000000000 +0000
+++ browser/app/profile/firefox.js.oden	2012-06-08 08:02:36.000000000 +0000
@@ -239,6 +239,9 @@
 pref("general.useragent.locale", "@AB_CD@");
 pref("general.skins.selectedSkin", "classic/1.0");
 
+// Use LANG environment variable to choose locale
+pref("intl.locale.matchOS", true);
+
 pref("general.smoothScroll", true);
 #ifdef UNIX_BUT_NOT_MAC
 pref("general.autoScroll", false);
