--- config/pythonpath.py.orig	2014-08-12 16:15:24.136152883 +0100
+++ config/pythonpath.py	2014-08-12 16:16:09.483149542 +0100
@@ -45,7 +45,7 @@
     frozenglobals['__name__'] = '__main__'
     frozenglobals['__file__'] = script
 
-    execfile(script, frozenglobals)
+    exec(compile(open(script, "rb").read(), script, 'exec'), frozenglobals)
 
 # Freeze scope here ... why this makes things work I have no idea ...
 frozenglobals = globals()
