--- mozilla-release/browser/components/shell/src/nsGNOMEShellService.cpp.appname	2013-04-10 18:21:58.743798117 +0200
+++ mozilla-release/browser/components/shell/src/nsGNOMEShellService.cpp	2013-04-10 18:23:18.455801580 +0200
@@ -113,7 +113,11 @@ nsGNOMEShellService::Init()
                    getter_AddRefs(appPath));
   NS_ENSURE_SUCCESS(rv, rv);
 
-  return appPath->GetNativePath(mAppPath);
+  /* This path to firefox is hardcoded for the default path setting */
+  mAppPath.Assign ("/usr/bin/firefox");
+
+  //return appPath->GetNativePath(mAppPath);
+  return NS_OK;
 }
 
 NS_IMPL_ISUPPORTS1(nsGNOMEShellService, nsIShellService)
