--- media/mtransport/third_party/nrappkit/nrappkit.gyp.orig	2013-01-15 12:38:17.315083776 +0000
+++ media/mtransport/third_party/nrappkit/nrappkit.gyp	2013-01-15 12:38:46.756082054 +0000
@@ -217,6 +217,7 @@
                      'RETSIGTYPE=void',
                      'TIME_WITH_SYS_TIME_H=1',
                      '__UNUSED__="__attribute__((unused))"',
+                     'NO_REG_RPC'
                  ],
 
 		 'include_dirs': [
