--- mozilla-release/browser/app/profile/firefox.js.disable-software-update	2012-11-21 19:23:57.000000000 +0400
+++ mozilla-release/browser/app/profile/firefox.js	2012-11-21 20:26:55.826335163 +0400
@@ -237,7 +237,7 @@ pref("general.useragent.override.natwest
 pref("general.useragent.override.raiffeisen.hu",         "Gecko/[^ ]*#Gecko/20100101"); // bug 795348
 
 // At startup, check if we're the default browser and prompt user if not.
-pref("browser.shell.checkDefaultBrowser", true);
+pref("browser.shell.checkDefaultBrowser", false);
 
 // 0 = blank, 1 = home (browser.startup.homepage), 2 = last visited page, 3 = resume previous browser session
 // The behavior of option 3 is detailed at: http://wiki.mozilla.org/Session_Restore
