--- content/media/plugins/MediaResourceServer.cpp.orig	2013-10-30 10:39:31.200013843 +0000
+++ content/media/plugins/MediaResourceServer.cpp	2013-10-30 10:39:36.424014118 +0000
@@ -228,7 +228,7 @@
       start = strtoll(s+byteRange.Length(), nullptr, 10);
 
       // Clamp 'start' to be between 0 and the resource length.
-      start = std::max(0ll, std::min(resource->GetLength(), start));
+      start = std::max(0ll, (long long)std::min(resource->GetLength(), start));
     }
   }
 
