--- mozilla-release/toolkit/mozapps/installer/packager.mk.nsspatch	2012-11-20 06:46:22.000000000 +0400
+++ mozilla-release/toolkit/mozapps/installer/packager.mk	2012-11-21 19:19:02.856402131 +0400
@@ -608,7 +608,7 @@ ifeq ($(OS_ARCH),OS2)
 NSS_DLL_SUFFIX	= .DLL
 SIGN_CMD	= $(MOZILLA_DIR)/toolkit/mozapps/installer/os2/sign.cmd $(DIST)
 else
-SIGN_CMD	= $(strip $(RUN_TEST_PROGRAM) $(_ABS_DIST)/bin/shlibsign$(BIN_SUFFIX) -v -i)
+SIGN_CMD	= $(strip $(RUN_TEST_PROGRAM) /bin/shlibsign$(BIN_SUFFIX) -v -i)
 endif
 endif
 
