--- mozilla-esr17/content/html/content/src/nsHTMLMediaElement.cpp.moz-ogg	2013-01-21 13:00:36.541853530 -0500
+++ mozilla-esr17/content/html/content/src/nsHTMLMediaElement.cpp	2013-01-21 13:00:46.527111470 -0500
@@ -2333,11 +2333,7 @@
 #endif
 #ifdef MOZ_OGG
   if (IsOggType(aType)) {
-#ifdef MOZ_GSTREAMER 
-    nsRefPtr<nsGStreamerDecoder> decoder = new nsGStreamerDecoder();
-#else
     nsRefPtr<nsOggDecoder> decoder = new nsOggDecoder();
-#endif
     if (decoder->Init(this)) {
       return decoder.forget();
     }
