diff -p -up ./browser/locales/generic/profile/mimeTypes.rdf.orig ./browser/locales/generic/profile/mimeTypes.rdf
--- ./browser/locales/generic/profile/mimeTypes.rdf.orig	2009-10-19 15:53:21.000000000 -0200
+++ ./browser/locales/generic/profile/mimeTypes.rdf	2009-10-19 15:54:17.000000000 -0200
@@ -10,4 +10,28 @@
       </Seq> 
     </NC:MIME-types> 
   </Description> 
-</RDF>
+
+  <RDF:Seq RDF:about="urn:mimetypes:root">
+  </RDF:Seq>
+  <RDF:Description RDF:about="urn:handler:local:/usr/bin/xdg-email"
+                   NC:prettyName="xdg-email"
+                   NC:path="/usr/bin/xdg-email" />
+  <RDF:Description RDF:about="urn:scheme:externalApplication:mailto"
+                   NC:prettyName="xdg-email"
+                   NC:path="/usr/bin/xdg-email" />
+  <RDF:Seq RDF:about="urn:schemes:root">
+    <RDF:li RDF:resource="urn:scheme:mailto"/>
+  </RDF:Seq>
+  <RDF:Description RDF:about="urn:mimetypes">
+    <NC:MIME-types RDF:resource="urn:mimetypes:root"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:scheme:handler:mailto"
+                   NC:alwaysAsk="false">
+    <NC:possibleApplication RDF:resource="urn:handler:local:/usr/bin/xdg-email"/>
+    <NC:externalApplication RDF:resource="urn:scheme:externalApplication:mailto"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:scheme:mailto"
+                   NC:value="mailto">
+    <NC:handlerProp RDF:resource="urn:scheme:handler:mailto"/>
+  </RDF:Description>
+</RDF:RDF>
