--- mozilla/modules/libpref/src/init/all.js.proxy	2006-08-24 20:39:23.000000000 +0200
+++ mozilla/modules/libpref/src/init/all.js	2006-08-24 20:41:18.000000000 +0200
@@ -52,7 +52,8 @@
 pref("general.useragent.locale", "chrome://navigator/locale/navigator.properties");
 pref("general.useragent.contentlocale", "chrome://navigator-region/locale/region.properties");
 
-pref("general.config.obscure_value", 13); // for MCD .cfg files
+pref("general.config.obscure_value", 0); // for MCD .cfg files
+pref("general.config.filename", "firefox.cfg");
 
 pref("browser.cache.disk.enable",           true);
 pref("browser.cache.disk.capacity",         51200);
