Index: mozilla-central/js/src/Makefile.in
===================================================================
--- mozilla-central.orig/js/src/Makefile.in
+++ mozilla-central/js/src/Makefile.in
@@ -923,7 +923,8 @@ endif
 # Needed to "configure" it correctly.  Unfortunately these
 # flags wind up being applied to all code in js/src, not just
 # the code in js/src/assembler.
-CXXFLAGS += -DUSE_SYSTEM_MALLOC=1 -DENABLE_ASSEMBLER=1 -DENABLE_JIT=1
+#CXXFLAGS += -DUSE_SYSTEM_MALLOC=1 -DENABLE_ASSEMBLER=1 -DENABLE_JIT=1
+CXXFLAGS += -DUSE_SYSTEM_MALLOC=1 -DENABLE_ASSEMBLER=0 -DENABLE_JIT=0 -DENABLE_YARR_JIT=0
 
 INCLUDES +=	-I$(srcdir)/assembler -I$(srcdir)/yarr
 
